### PR TITLE
add cask for kui shell 0.1.2

### PR DESCRIPTION
**Pull request by docwhat**
seee also #66233

After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] The commit message includes the cask’s name and version.
- [x] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

Additionally, if **adding a new cask**:

- [x] Named the cask according to the [token reference].
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.
- [x] Checked there are no [open pull requests] for the same cask.
- [x] Checked the cask was not [already refused].
- [x] Checked the cask is submitted to [the correct repo].

[token reference]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md
[open pull requests]: https://github.com/Homebrew/homebrew-cask/pulls
[already refused]: https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues
[the correct repo]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask
[version-checksum]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/sha256.md#updating-the-sha256

### CODE REVIEW

File: Casks/kui.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+cask 'kui' do
+  version '0.1.2'
+  sha256 '124d713f59ab8d065040106781dd9b9f2163c849e6cbc0204a91e59c117bd0f4'
+
+  url "https://github.com/kui-shell/plugin-kubeui/releases/download/#{version}/Kui-darwin-x64.tar.bz2"
```
Comment (vitorgalvao) [inline Casks/kui.rb]: This and the `homepage` are different repos and do not seem to point to the same thing. Furthermore, this one does not pass our notability requirements (it’s under 50 star).

**Comment by ran-dall** [comment]
@docwhat This is how I'd do this...

```ruby
cask 'kui' do
  version '5.5.6'
  sha256 'f5a3456231db6eebd97275df7cff33fadfa82368681ff0500d5ff084953772ae'

  # s3-api.us-geo.objectstorage.softlayer.net/kui-dev was verified as official when first introduced to the cask
  url "https://s3-api.us-geo.objectstorage.softlayer.net/kui-dev/Kui-darwin-x64.tar.bz2"
  appcast 'https://github.com/IBM/kui/releases.atom'
  name 'Kui'
  homepage 'https://github.com/IBM/kui'

  app 'Kui-darwin-x64/Kui.app', target: 'Kui.app'
end
```

@vitorgalvao I know the `app 'Kui-darwin-x64/Kui.app', target: 'Kui.app'` might not be correct, but I would much rather have the `.app` than another folder.

**Comment by vitorgalvao** [comment]
> I know the `app 'Kui-darwin-x64/Kui.app', target: 'Kui.app'` might not be correct, but I would much rather have the `.app` than another folder.

`app 'Kui-darwin-x64/Kui.app'` only copies `Kui.app`, it does not replicate the path where it took it from. `app 'Kui-darwin-x64/Kui.app', target: 'Kui.app'` and `app 'Kui-darwin-x64/Kui.app'` have the same exact effect (so the latter is preferred).

**Comment by ran-dall** [comment]
@docwhat Here's the updated recommendation...
```ruby

```ruby
cask 'kui' do
  version '5.5.6'
  sha256 'f5a3456231db6eebd97275df7cff33fadfa82368681ff0500d5ff084953772ae'

  # s3-api.us-geo.objectstorage.softlayer.net/kui-dev was verified as official when first introduced to the cask
  url "https://s3-api.us-geo.objectstorage.softlayer.net/kui-dev/Kui-darwin-x64.tar.bz2"
  appcast 'https://github.com/IBM/kui/releases.atom'
  name 'Kui'
  homepage 'https://github.com/IBM/kui'

  app 'Kui-darwin-x64/Kui.app'
end
```
```

**Comment by docwhat** [comment]
Why use the S3 url for the US? 

In fact, why not use the original IBM url?

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by ran-dall** [comment]
FME sometime those links don't work correctly and the redirect get's wonky. Otherwise, I'd defer to @vitorgalvao for his insight 🙏

**Comment by vitorgalvao** [comment]
When a link redirects, we tend to prefer the redirect. That said, in this case both seem to be unversioned and the pre-redirection link is on the same host, so it may make more sense.

**Comment by docwhat** [comment]
`https://s3-api.us-geo.objectstorage.softlayer.net/kui-dev/Kui-darwin-x64.tar.bz2` seems to be an older version from Nov. 27th vs. Dec. 3rd.

It looks like I'm submitting this just as they are restructuring their project into a generic "kui-shell" with plugins for various things.

*sigh*

**Comment by vitorgalvao** [comment]
> It looks like I'm submitting this just as they are restructuring their project into a generic "kui-shell" with plugins for various things.

Do you prefer to close and reopen when everything is in place?

**Comment by docwhat** [comment]
Yeah. 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Pull request closed without merging**